### PR TITLE
Adding Charge and Staff of Healing to the spellbook + category reshuffle

### PR DESCRIPTION
**Pull request by Incoming5643**
The Staff of Healing is now selectable for a cost of 1 in the spellbook

The spell "Charge" is now selectable for a cost of 2 in the spellbook

The spellbook's category system has been reworked to help deal with the huge scroll bars it previously had and help identify to wizards what needs spells and items serve. The spellbook window is also larger by default.

The new categories are:

Offensive:
Fireball
Disintegrate
Blind
Mutate
Flesh to Stone
Lightning Bolt
Curse of the Barnyard
Staff of Change
Staff of Chaos

Defensive:
Magic Missile
Disable Tech
Repulse
Stop Time
Smoke
Forcewall
Bind Soul
Staff of Healing
Scrying Orb
Wand Assortment
Armor Set

Mobility:
Mind Transfer
Blink
Teleport
Ethereal Jaunt
Knock
Staff of Door Creation

Assistance:
Instant Summons
Charge
Staff of Animation
Soul Stone Shards + Artificer
Necromantic Stone
Apprentice Contract
Bottle of Blood

Fixes an unreported bug where targeting your mouth with a staff of healing allowed self healing through suicide.

Adds an easter egg to clever wizards who try to charge the spellbook to get more charges.

### CODE REVIEW

File: code/datums/spells/charge.dm [hunk 1]
```diff
@@ -35,14 +35,19 @@
 					charged_item = M
 					break
 			else if(istype(item, /obj/item/weapon/spellbook/oneuse))
-				var/obj/item/weapon/spellbook/oneuse/I = item
-				if(prob(80))
-					user.visible_message("<span class='warning'>[I] catches fire!</span>")
-					qdel(I)
+				if(istype(item, /obj/item/weapon/spellbook/oneuse))
```
Comment (RemieRichards) [inline code/datums/spells/charge.dm]: you're istyping this twice... stop that, that's bad.

Comment (Xhuis) [inline code/datums/spells/charge.dm]: DOUBLE THE SANITY

**Comment by Xhuis** [comment]
I'm not so sure about giving them Charge.

File: code/datums/spells/charge.dm (hunk 1 above)
Comment (Incoming5643) [inline code/datums/spells/charge.dm]: You can never be TOO careful.

Will fix.

**Comment by Incoming5643** [comment]
To be fair you can already just give a healing apprentice a wand belt and get the same effect for the same point cost.

**Comment by PKPenguin321** [comment]
:+1: i've always wanted this

**Comment by Lumipharon** [comment]
charge decreases the max charge of the item you're charging afaik, so evenif you have a wand belt you'll quickly render them useless.

Staff of healin is bad if you can self heal (Haven't been able to afaik, but some people sayin you can now?), other wise it's ok.

**Comment by Zilenan91** [comment]
Adding charge will make it so wizards raiding the armory won't be impotent in high pop, so this is good by me.

**Comment by Bawhoppen** [comment]
@Lumipharon You can now with staffs AND wands, but you can also get a wand of healing from a wand belt aswell

**Comment by Lumipharon** [comment]
wand has finite charge (even with the charge spell), so it's a bit different.

**Comment by Supermichael777** [comment]
i think you can elf heal with the staff by shooting yourself in the face. not sure though(why are magic stalfs guns?)

**Comment by Incoming5643** [comment]
@Supermichael777 That's fixed in this pull.

File: code/datums/spells/charge.dm [hunk 2]
```diff
@@ -35,14 +35,19 @@
...
+					var/obj/item/weapon/spellbook/oneuse/I = item
+					if(prob(80))
+						user.visible_message("<span class='warning'>[I] catches fire!</span>")
+						qdel(I)
```
Comment (Dorsisdwarf) [inline code/datums/spells/charge.dm]: You can probbably actually set them on fire now

File: code/modules/projectiles/guns/magic/staff.dm [hunk 3]
```diff
@@ -25,6 +25,9 @@
 	icon_state = "staffofhealing"
 	item_state = "staffofhealing"
 
+/obj/item/weapon/gun/magic/staff/healing/handle_suicide() //Stops people trying to commit suicide to heal themselves
```
Comment (Dorsisdwarf) [inline code/modules/projectiles/guns/magic/staff.dm]: Make it dust the user instead, Plane of Positive Energy-style

**Comment by Incoming5643** [comment]
I'm gonna hold off on having books actually catch fire for now, as there's some bugs to items on fire in your UI that I'd rather not deal with.

Beyond that I feel pretty confident that this is ready

**Comment by Zilenan91** [comment]
I greatly look forward to having charge in my wizard arsenal

**Comment by Razharas** [comment]
@Incoming5643

**Comment by Incoming5643** [comment]
I'm not seeing any outstanding issues here

**Comment by KorPhaeron** [comment]
@Cheridan 
Can this get merged so we can touch the spellbook again?

**Pull request merged**